src/base/utils/io.cpp

```diff
@@ -89,13 +89,34 @@ nonstd::expected<QByteArray, Utils::IO::ReadError> Utils::IO::readFile(const Pat
         return nonstd::make_unexpected(ReadError {ReadError::ExceedSize, message});
     }
 
-    // Do not use `QIODevice::readAll()` it won't stop when reading `/dev/zero`
-    const QByteArray data = file.read(fileSize);
-    if (const qint64 dataSize = data.size(); dataSize != fileSize)
+    const qint64 targetSize = (maxSize < 0) ? fileSize : std::min(fileSize, maxSize);
+
+    QByteArray data;
+    data.resize(targetSize);
```

Comment:
Maybe create it of the desired size instead of create+resize?

Reply:
https://doc.qt.io/qt-6/qbytearray.html#QByteArray-3
It is not available in Qt 6.2 or 5.15.

Reply:
> https://doc.qt.io/qt-6/qbytearray.html#QByteArray-3
> It is not available in Qt 6.2 or 5.15.

I didn't notice any mention of this in the documentation.
BTW, it is already used in the code, so at least Qt 5.15 has no problems with it.

Reply:
>I didn't notice any mention of this in the documentation.

Its not listed in the older version of the documentation. 
Anyway if CI doesn't complain then I have no problems.